### ISSUE TITLE
bindParam() is deprecated and you should use bindParameters() instead

### ISSUE THREAD

**Issue by alexandrahrastnik** [IBM/nodejs-idb-pconnector#135]
**Is your feature request related to a problem? Please describe.**
Deprecation message
The following message shows up on execution of `DBPool:prepareExecute`:
> Statement deprecated As of 1.1.0, bindParam() is deprecated and you should use bindParameters() instead. node_modules/idb-pconnector/lib/dbPool.js:272:23

The "problem": The deprecation message is written to the stderr stream and blocks CI pipelines, if configured to count messages on stderr as errors. Easy to work around by altering the CI pipeline configuration, but not ideal.

**Describe the solution you'd like**
Refactor `lib/dbPool.js` to use `bindParameters` instead of `bindParam`.

**Comment by abmusse** [comment]
The module we use to print the deprecation messages is https://www.npmjs.com/package/depd. 

Maybe there is a way to configure the message to be logged to a different output?

One way might be to use this:

https://github.com/dougwilson/nodejs-depd#processondeprecation-fn

**Comment by kadler** [comment]
That might be a good workaround, but ideally we shouldn't be using deprecated methods internally (except inside methods that are deprecated themselves).

**Comment by abmusse** [comment]
Thats true. The right solution is to switch to bindParameters internally.

**Comment by abmusse** [comment]
prepareExecute accepts and options object to set on the parameters. Here are the option values it would accept.
https://github.com/IBM/nodejs-idb-pconnector/tree/master/docs#dbpoolprepareexecutesql-params-options
```
value: String | Number | Buffer | boolean | null
io: in | out | both
asClob: true | false
```

The incompatibility is that bindParameters determines the io and data type internally. So by switching to bindParameters we would need to stop caring about the options object. Which is a good thing but a breaking change.

**Comment by kadler** [comment]
> Also parameters can be customized at an individual level by passing an object within the params Array.

Looks like a lot of this function requires using the old API. Not sure if it would be best to just mark it deprecated and add a new API which calls bindParam or have extra code internally that will call bindParam if no configuration is set and fall back to bindParameters if the user passed configuration options. The first option is much simpler for the user to understand "the whole function is deprecated" vs "it's deprecated if you do this... or this... or this...", but then we have to come up with a new name and get users to migrate to the new function.

**Comment by abmusse** [comment]
Yeah prepareExecute is tightly coupled with using bindParams. Simply deprecating it and create a new function that uses bindParameters would be more understandable solution.

> then we have to come up with a new name and get users to migrate to the new function.

Thats the challenging part. I'm not sure what to call the new function and most of the code would be the same.

**Comment by alexandrahrastnik** [comment]
Hey there, 
it's been some time since the last activity on this question / issue. Are there any plans on a future release of idb-pconnector with a new interface using non-deprecated methods?

**Comment by abmusse** [comment]
Hi @patrickhrastnik ,

I think the approach would be:

1) creating a new method that uses bindParameters 
2) Then deprecate prepareExecute.

This should be relatively straight forward to implement.
Any suggestions on what the new method should be named?

We would be open to accepting PR with the new method if this is something you could help contribute.
